daemon/src/payout_curve/csr_tools.rs

```diff
@@ -89,6 +92,9 @@ impl CSR {
 }
 
 fn lu_solve(a: &Array2<f64>, b: &Array1<f64>) -> Result<Array1<f64>, Error> {
+    if a.shape()[0] != a.shape()[1] {
+        return Result::Err(Error::MatrixMustBeSquare);
+    }
```

Comment:
Are we calling `lu_solve` somewhere outside of `matrix_solve`? I think we should only have one check.